packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -70,11 +123,150 @@ - (void)handleProductRequestMethodCall:(FlutterMethodCall *)call result:(Flutter
                                  details:call.arguments]);
       return;
     }
+    for (SKProduct *product in response.products) {
+      [self.productsCache setObject:product forKey:product.productIdentifier];
+    }
     result([FIAObjectTranslator getMapFromSKProductsResponse:response]);
     [weakSelf.requestHandlers removeObject:handler];
   }];
 }
 
+- (void)createPaymentWithProductID:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSString class]]) {
+    result([FlutterError
+        errorWithCode:@"storekit_invalide_argument"
+              message:@"Argument type of createPaymentWithProductID is not a string."
+              details:call.arguments]);
+    return;
+  }
+  NSString *productID = call.arguments;
+  SKProduct *product = [self.productsCache objectForKey:productID];
+  if (!product) {
+    result([FlutterError
+        errorWithCode:@"storekit_product_not_found"
+              message:@"Cannot find the product. To create a payment of a product, you must query "
+                      @"the product with SKProductRequestMaker.startProductRequest."
+              details:call.arguments]);
+    return;
+  }
+  SKPayment *payment = [SKPayment paymentWithProduct:product];
+  [self.paymentsCache setObject:payment forKey:productID];
+  result([FIAObjectTranslator getMapFromSKPayment:payment]);
+}
+
+- (void)addPayment:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSDictionary class]]) {
+    result([FlutterError errorWithCode:@"storekit_invalide_argument"
+                               message:@"Argument type of addPayment is not a map"
+                               details:call.arguments]);
+    return;
+  }
+  NSDictionary *paymentMap = (NSDictionary *)call.arguments;
+  NSString *productID = [paymentMap objectForKey:@"productID"];
+  SKPayment *payment = [self.paymentsCache objectForKey:productID];
+  // User can use  payment object with mutable = true and add simulatesAskToBuyInSandBox = true to
+  // test the payment flow.
+  if (!payment || [paymentMap[@"mutable"] boolValue] == YES) {
```

Comment:
This ends up creating a mutable payment whenever payment isn't specified, not just when `mutable` is explicitly specified as an option. Is that something we want to do?

Reply:
The mutable keyword is indeed confusing. I created this for user to test the payment in sandbox, I have updated the mutable keyword to `usePaymentObject` in a different branch. Do you think changing the keyboard to `usePaymentObject` makes more sense here?

Reply:
Yeah, I think that makes sense.

I'm still a little confused over the code silently creating an `SkMutable` payment object whenever it can't find a pre-existing payment in the map. Do we want to default to that or would it make sense to error if we can't find an existing payment instead?

Reply:
Oh you are right. Going to update. I am also going to re arrange the code a bit to make it more readable.